src/app/analysis-types-list/analysis-types-list.service.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * analysis-types-list.module.ts
+ * Module for analysis types list.
+ *
+   Copyright 2018 Roman Lisak
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/Rx';
+import 'rxjs/Rx';
+
+@Injectable()
+export class AnalysisTypesListService {
+
+  constructor(private client: HttpClient) { }
+
+  getAlgorithms(algorithmsUrl: string): Observable<any> {
```

Comment:
I think that construction of the URL should be inside this service, not in the component.